test/TestGrains/MethodInterceptionGrain.cs

```diff
@@ -97,6 +98,53 @@ public Task<object> Invoke(MethodInfo methodInfo, InvokeMethodRequest request, I
         public Task<string> GetInputAsString(bool input) => Task.FromResult(input.ToString(CultureInfo.InvariantCulture));
 
         public Task<int> GetBestNumber() => Task.FromResult(38);
+    }
+
+    public class GrainCallFilterTestGrain : Grain, IGrainCallFilterTestGrain, IGrainCallFilter, IGrainInvokeInterceptor
+    {
+        private const string Key = GrainCallFilterTestConstants.Key;
+        private IGrainCallContext context;
+
+        public async Task<string> Execute(bool early, bool mid, bool late)
+        {
+            if (late)
+            {
+                this.context.Arguments[2] = false;
+                await this.context.Invoke();
```

Comment:
I understand you are probably showing how to break this, but this is really not how filters should be used, right? And since it's the only example of usage in a grain, it might be misleading. Do you want to create a separate test/example grain that doesn't do all these edgy stuff and just plays along with the happy path?